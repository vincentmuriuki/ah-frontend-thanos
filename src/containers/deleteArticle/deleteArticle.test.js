import React from 'react';
import { mount, shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import DeleteArticlePage, { mapDispatchToProps } from './index';

describe('<CreateArticlePage />', () => {
  let DeleteArticlePageComponent;
  let wrapper;
  beforeEach(() => {
    jest.resetModules();
    const middlewares = [thunk];
    const mockStore = configureStore(middlewares);
    const initialState = { deleteArticleReducer: { articleDelete: { message: 'Article deleted succesfully' } } };
    const store = mockStore(initialState);
    const props = { articleId: 1 };
    DeleteArticlePageComponent = mount(
      <DeleteArticlePage article={jest.fn()} store={store} articleId={props.articleId} />,
    );
    wrapper = shallow(
      <DeleteArticlePage store={store} articleId={props.articleId} />,
    );
  });

  it('should render the component', () => {
    expect(DeleteArticlePageComponent).toMatchSnapshot();
  });

  it('should dispatch a method to get user input', () => {
    const dispatch = jest.fn();
    mapDispatchToProps(dispatch).articleDelete({});
  });

  test('Test function', () => {
    const event = { preventDefault: jest.fn() };
    expect(wrapper.dive().instance().handleSubmit(event));
  });

  it('should test user data without fail', () => {
    DeleteArticlePageComponent
      .find('button')
      .simulate('submit');
  });
});
