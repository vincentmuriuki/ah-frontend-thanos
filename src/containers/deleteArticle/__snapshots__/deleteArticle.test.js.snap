// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateArticlePage /> should render the component 1`] = `
<Connect(DeleteArticlePage)
  article={[MockFunction]}
  articleId={1}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <DeleteArticlePage
    article={[MockFunction]}
    articleDelete={[Function]}
    articleId={1}
    deleteArticleReducer={
      Object {
        "articleDelete": Object {
          "message": "Article deleted succesfully",
        },
      }
    }
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      }
    }
  >
    <div>
      <DeleteArticle
        onClick={[Function]}
      >
        <div>
          <button
            className="btn btn-danger"
            onClick={[Function]}
            type="submit"
          >
            DELETE ARTICLE
          </button>
        </div>
      </DeleteArticle>
    </div>
  </DeleteArticlePage>
</Connect(DeleteArticlePage)>
`;
